src/bc/de.rs

```diff
@@ -116,99 +116,100 @@ fn bc_modern_msg<'a, 'b>(
         Err,
     };
 
-    let mut in_bin_mode = context.in_bin_mode.contains(&header.msg_id);
+    let ext_len = match header.payload_offset {
+        Some(off) => off,
+        _ => 0, // If missing payload_offset treat all as payload
+    };
 
-    // We'd like to know where the XML stops, but we haven't parsed the XML yet to see if the
-    // binaryData offset in the header is valid
-    let end_of_xml = if in_bin_mode {
-        0
-    } else {
-        match header.bin_offset {
-            Some(off) if off > 0 => off,
-            _ => header.body_len,
+    if header.msg_id == 1 && (header.response_code >> 8) == 0xdd {
+        // Login reply has the encryption info
+        // Set that the encryption type now
+        let encryption_protocol_byte = (header.response_code & 0xff) as usize;
+        match encryption_protocol_byte {
+            0x00 => context.set_encrypted(EncryptionProtocol::Unencrypted),
+            0x01 => context.set_encrypted(EncryptionProtocol::BCEncrypt),
+            0x03 => context.set_encrypted(EncryptionProtocol::Aes(AesKey {
+                nonce: None,
+                passwd: None,
+            })),
+            _ => context.set_encrypted(EncryptionProtocol::Unknown),
```

Comment:
Return an error here, don't continue and explode (`unimplemented`) later.  You can remove the Unknown type since nothing meaningfully handles that case.

Reply:
I fixed this, see my branch `protocol_updates_aeskey`.

Reply:
Pulled in